spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/pulsar/PulsarContainerPropertiesCustomizer.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2012-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.pulsar;
+
+import org.springframework.pulsar.listener.PulsarContainerProperties;
+
+/**
+ * The interface to customize a {@link PulsarContainerProperties}.
+ *
+ * @author Chris Bono
+ * @since 3.2.0
+ */
+@FunctionalInterface
+public interface PulsarContainerPropertiesCustomizer {
```

Comment:
I was originally going to put these container props customizers in the framework. However, the container factories do not need to accept a customizer as they already accept the container props. Non-boot users will already have to create the container props in order to create the container factory. They will therefore just adjust the props accordingly - no need to also have a customizer. 

IOW - the framework does not need the concept of container props customizers.